src/dfx/src/commands/canister/delete.rs

```diff
@@ -81,25 +81,35 @@ async fn delete_canister(
     withdraw_cycles_to_dank_principal: Option<String>,
 ) -> DfxResult {
     let log = env.get_logger();
-    let mut canister_id_store = CanisterIdStore::for_env(env)?;
-    let canister_id =
-        Principal::from_text(canister).or_else(|_| canister_id_store.get(canister))?;
+    let mut canister_id_store =
+        CanisterIdStore::for_env(env).context("Failed to load canister store.")?;
+    let canister_id = Principal::from_text(canister)
+        .or_else(|_| canister_id_store.get(canister))
+        .with_context(|| format!("Failed to read canister id for {}.", canister))?;
     let mut call_sender = call_sender;
     let to_dank = withdraw_cycles_to_dank || withdraw_cycles_to_dank_principal.is_some();
 
     // Get the canister to transfer the cycles to.
     let target_canister_id = if no_withdrawal {
         None
     } else if to_dank {
-        Some(Principal::from_text("aanaa-xaaaa-aaaah-aaeiq-cai")?)
+        Some(
+            Principal::from_text("aanaa-xaaaa-aaaah-aaeiq-cai")
+                .context("Failed to read principal aanaa-xaaaa-aaaah-aaeiq-cai.")?,
+        )
```

Comment:
Another option here would be to use this form (example from elsewhere) to construct the principal constant, so no error is possible:
```
pub const MAINNET_CYCLE_MINTER_CANISTER_ID: Principal =
    Principal::from_slice(&[0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x04, 0x01, 0x01]);
```

Otherwise, I'd suggest including the term "dank" in the context message.